lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb

```diff
@@ -62,6 +63,10 @@ def validate_options
             unless options['mode'].present? && modes.include?(options['mode'])
                 errors.add(:base, "mode is a required field and must be one of: #{modes.join(', ')}")
             end
+
+            unless options['not_purchasable_format_list'].is_a?(Array)
+                errors.add(:base, 'not_purchasable_format_list is a required field')
+            end
```

Comment:
@Jhalter5Stones I would make sure the error message actually reflects the validation.

Additionally, I would make sure that you only check whether the option is an array if the option is provided

Reply:
```
if options[...].present? && !options[...].is_a?(Array)
    errors.add( ... )
end
```